### ISSUE TITLE
Whats different with vite-plugin-federation

### ISSUE THREAD

**Issue by AndreiSoroka** [module-federation/vite#4]
Hi there!
Can you provide a detailed description of what your differences with https://github.com/originjs/vite-plugin-federation

Like this:

| Feature | MF Vite | vite-plugin |
|---|---|---|
| Support Vite-Webpack | yes | yes |
| Support Webpack-Vite | yes | yes |
| feature 3 | yes | yes |
| feature 4 | yes | yes |

My situation: in the remote host app I connected a lot of different types of apps (and webpack and vite). I have been using the federation module for a very long time in my application, so the migration will be very painful. Backward compatibility is just as important to me.

**Comment by gioboa** [comment]
This plugin is using a configuration really similar to the webpack one. If your migration is painful I think you can remain with the plugin you're using right now

**Comment by gioboa** [comment]
@AndreiSoroka do I answer to your question?

**Comment by echjordan0** [comment]
Thank so much for this plugin. Does your answer mean that migration of all (host and remote(s)) to Vite is required to use this plugin? Is it possible to make a host using native federation compatible with a webpack remote?

**Comment by echjordan0** [comment]
Sorry, your comment is a bit ambiguous. Is it meant to say "you should **not** mix and match webpack and vite applications"?

**Comment by gioboa** [comment]
You can use webpack module federation

**Comment by echjordan0** [comment]
Would you be able to provide an example of Vue2 vite host to webpack remote? I have been using [this example](https://github.com/module-federation/vite/issues/5) but with a webpack remote and I'm not sure how to generate the remote manifest required to provide to initFederation (as in this [example](https://www.npmjs.com/package/@softarc/native-federation#initializing-a-host)) on the host side

**Comment by gioboa** [comment]
Can you create a basic example pls? I will check it out

**Comment by echjordan0** [comment]
I'm not able to provide one at the moment -- I'm going to proceed with `@originjs/vite-plugin-federation` to unblock us but I would like to use this plugin eventually. If I have time I will open a new issue requesting an example and providing basic setup. Thank you so much for your quick responses.

**Comment by kaidjohnson** [comment]
In case it helps someone, it is _not_ possible to use Native Federation (eg this Vite plugin) with Webpack Module Federation plugin. Native Federation operates via `remoteEntry.json` (config), whereas Module Federation operates on `remoteEntry.js` (code) and they are [not interoperable](https://github.com/angular-architects/module-federation-plugin/issues/385#issuecomment-1826188096). 

`@originjs/vite-plugin-federation` is a vite plugin for Module Federation that is compatible with Webpack remotes. MF Vite (this plugin) is for Native Federation and must be used with other Native Federation remotes. Per the original question, I believe the feature table looks something like:

Feature | MF Vite | vite-plugin
-- | -- | --
Support Vite-Webpack | no | yes
Support Webpack-Vite | no | yes
Support Vite-Native Federation | yes | no
Support Native Federation-Vite | yes | no

**Comment by gioboa** [comment]
@kaidjohnson Thanks for your clarification. 

> yes, you should mix and match webpack and vite applications.

So my previous assumption was wrong 😔😅

**Comment by echjordan0** [comment]
I'll chip in that another blocker for us to using this plugin was that we found we had to reconfigure much of the esbuild set up of the remote rather than the plugin inheriting configuration from Vite. This felt redundant/cumbersome. @kaidjohnson Curious if you run into the same thing or found a workaround for it prior to running into the webpack incompatibility?

**Comment by kaidjohnson** [comment]
@gioboa Yes, I have created a [basic example ](https://github.com/kaidjohnson/federation-interop) demonstrating the issue with interoperability between webpack Module Federation and vite Native Federation, both with a vite host and webpack remote as well as a webpack host with a vite remote. The issues are slightly different, depending on which side each stack is on.

It mostly comes down to usage of `remoteEntry.json` (Native Federation) vs `remoteEntry.js` (Module Federation). If Native Federation could provide tooling to (optionally) generate a compatible `remoteEntry.js` and could consume `remoteEntry.js`, migration towards the new standard (Native Federation) would be much improved. Let me know if the examples make sense - they are pretty bare bones, but definitely highlight the problems.

**Comment by Cafezinho** [comment]
Anyone figured out a way to handle remoteEntry.json with a Module Federation host?

**Comment by husayt** [comment]
@gioboa , @ScriptedAlchemy it will be really useful to add clarification to the description of this project, as current name is really misleading and confusing.

Native Federation and Module Federation are two separate implementations of modular architecture pattern. Currently there is not even an example I know of which allows NF host to load MF remote of vice verse. Hence we can say they are incompatible.

This library is a Vite wrapper (very useful indeed) around [Softarc's Native Federation implementation](https://www.npmjs.com/package/@softarc/native-federation) . But when people come here they expect to see Module Federation plugin for Vite, i.e. something like [Originjs module federation plugin](https://github.com/originjs/vite-plugin-federation), which is in itself another tragedy due to maintainers seemingly abandoning the project.

This project is really important and useful, but naming needs to reflect what it really presents.

Thank you for all the amazing effort

**Comment by zhangHongEn** [comment]
I am using @module-federation/enhanced/runtime to rewrite the module-federation-vite plugin to solve:
1. originjs/vite-plugin-federation does not support dev mode
2. @module-federation/vite current plugin configuration is too complicated, I have never started it successfully
3. Using @module-federation/enhanced/runtime can solve the problem of remoteEntry.json and remoteEntry.js
4. Make vite support remote ts, module-federtion chrome plugin

**Comment by zhangHongEn** [comment]
the demo has been completed, dev and build are normal, the plugin will be completed this week, and I hope to have official cooperation
https://github.com/zhangHongEn/module-federation-vite

**Comment by gioboa** [comment]
Hi @zhangHongEn that's amazing 🤩 
Would you like to open a PR for that here please?
I'm more than happy to test and merge your contribution 👏

**Comment by zhangHongEn** [comment]
Yes, I will convert the manual configuration of this demo into a plugin this week and then make a pr

**Comment by zhangHongEn** [comment]
Almost done, npm install module-federation-vite -D, I temporarily published it in this package, it is already available
### There is one more problem to solve and I will raise a PR:
1. I need to be able to obtain the status of host init completion to ensure that the remotes entry has been loaded when the host obtains the shared. Now I don’t have a method and runtime hooks seem to be ineffective (https://github.com/zhangHongEn/module-federation-vite/blob/f11adc79f203ba5d7a0622a460c74a6f5e557489/packages/module-federation-vite/src/index.js#L86)
### There are still some functions that need to be enhanced:
1. RemoteEntry.js has been generated, and the ability to generate manifest.json needs to be added

**Comment by zhangHongEn** [comment]
#28
